rustc_codegen_ssa feels permanent enough not to need date-check (#1606)
rust-lang/rustc-dev-guide: src/backend/backend-agnostic.md

<!-- toc -->

> [`rustc_codegen_ssa`]
> provides an abstract interface for all backends to implement,
> to allow other codegen backends (e.g. [Cranelift]).

Reviewer: While we're at it, what do you think about changing the wording to this?

```suggestion
[`rustc_codegen_ssa`]
provides an abstract interface for all backends to implement,
enabling the creation of new backends (e.g. [Cranelift]).
```

Author: how about "to allow code sharing (e.g. with [Cranelift])"

Author: Actually that doesn't sound right. I think we should just remove the part starting with "enabling", or just mention all 3 backends?

Author: see if this is resolved with https://github.com/rust-lang/rustc-dev-guide/pull/1606/commits/dfb468a52fb606b6647e7240756f2c399654f179

> [Cranelift]: https://github.com/bytecodealliance/wasmtime/tree/HEAD/cranelift

Reviewer: What do you think about changing this to link to Rust's cranelift backend rather than the wasmtime repo? I think in its existing state it might be a bit confusing.

Author: I thought that would be a better link target as well. I do it.

Author: see if this is resolved with https://github.com/rust-lang/rustc-dev-guide/pull/1606/commits/dfb468a52fb606b6647e7240756f2c399654f179

> [`rustc_codegen_ssa`]: https://doc.rust-lang.org/stable/nightly-rustc/rustc_codegen_ssa/index.html

Reviewer: Might be better to link to the latest docs.

```suggestion
[`rustc_codegen_ssa`]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_codegen_ssa/index.html
```
